sources/programs/fizz-buzz/gnu-make/how-to-implement-the-solution.md

```diff
@@ -0,0 +1,397 @@
+* Is divisible by (`IS_DIVISIBLE_BY`)
+
+#### INC Function
+
+Let's look at the `INC` function first since it is the simplest, and it is
+used elsewhere. It takes a single argument which is a number represented as
+`x`'s and returns the number plus one represented as `x`'s. Here the function:
+
+```make
+INC = $(strip $(1) $(ONE))
+```
+
+Incrementing just involves append a space and a single `x` (the value of
+`ONE`) to the passed in argument. The [strip][4] function removes leading
+and trailing whitespace. This is needed for the case whether the argument has
+the value of zero, which is an empty string. Without this, ` x` would be
+returned instead of just `x`.
+
+#### IS_LESS_THAN Function
+
+The `IS_LESS_THAN` function takes a two arguments which are numbers
+represented as `x`'s and returns a `$(ONE)` (which is `x`) as a true value,
+and `$(ZERO)` (which is empty) as a false value. Here is the function:
+
+```make
+IS_LESS_THAN = $(if $(wordlist $(words $(call INC,$(1))),$(words $(2)),$(2)),$(ONE),$(ZERO))
+```
+
+Whoa! That's a lot to digest here! Let's break it down. Effectively, the way
+this works if to remove each item in the first argument from the second until
+the first or second argument is exhausted, whichever comes first. When the
+first argument is less than the second, a non-empty list will remain, and
+`$(ONE)` is returned. Otherwise, it will be an empty list, and `$(ZERO)` is
+returned.
+
+Let's work through a couple of examples. Let `ARG1` be `3` and `ARG2` be `5`,
+each represented as `x`'s:
+
+```make
+ARG1 := x x x
+ARG2 := x x x x x
+RESULT := $(call IS_LESS_THAN,$(ARG1),$(ARG2))
+```
+
+For this case `$(words $(call INC,$(ARG1)))` is `4` (`3 + 1`), and
+`$(words $(ARG2))` is `5`, so `$(wordlist 4,5,$(ARG2))` can be visualized as
+the right-hand side of this:
+
+```
+1 2 3 | 4 5
+x x x | x x
+```
+
+Since the result is non-empty, `$(ONE)` is returned.
+
+Let's work through the reverse of the previous example:
+
+```make
+ARG1 := x x x x x
+ARG2 := x x x
+RESULT := $(call IS_LESS_THAN,$(ARG1),$(ARG2))
+```
+
+For this case `$(words $(call INC,$(ARG1)))` is `6`, and `$(words $(ARG2))` is
+`3`, so `$(wordlist 6,3,$(ARG2))` can be visualized as the right-hand side of
+this:
+
+```
+1 2 3 | 4 5 6
+x x x |
+```
+
+Since the result is empty, `$(ZERO)` is returned.
+
+#### IS_DIVISIBLE_BY Function
+
+The `IS_DIVISIBLE_BY` function take two arguments:
+
+* Dividend represented as `x`'s
+* Divisor represented as `x`'s
+
+If the dividend is divisible by the divisor, `$(ONE)` is returned. Otherwise,
+`$(ZERO)` is returned. Here is the function:
+
+```make
+IS_DIVISIBLE_BY = $(if $(strip $(subst $(2),,$(1))),$(ZERO),$(ONE))
+```
+
+While this isn't as scary as the `IS_LESS_THAN` function, there is an
+interesting trick at work here. The [subst][8] take three arguments:
+
+1. The search string
+2. The replace string
+3. The string to modify
+
+How this is being used is to remove all instances of the divisor from the
```

Comment:
Some awkward phrasing here as well. I don't know exactly how I would rephrase it, but maybe something like "This is used to remove all instances of the divisor..."